### PR TITLE
[joyop] update dynamic configuration

### PR DESCRIPTION
**Pull request by haryngod**
Update joystick update dynamic configuration.

### CODE REVIEW

File: yocs_joyop/src/joyop.cpp [hunk 1]
```diff
@@ -246,6 +255,13 @@ void JoyOp::publish()
   }
 }
 
+void JoyOp::reconfigCB(gopher_core_msgs::BagJoystickConfig &config, uint32_t level) 
+{
+  ROS_INFO("Reconfigure joystick linear : %f, angular : %f", config.linear_scale, config.angular_scale);
```
Comment (stonier) [inline yocs_joyop/src/joyop.cpp]: Any `ROS_INFO` comments should follow the format used in the rest of this cpp code, i.e.

```
ROS_INFO("JoyOp: ...")
```

File: yocs_joyop/src/joyop.cpp [hunk 2]
```diff
@@ -41,6 +44,9 @@ class JoyOp
   bool enable_pressed_, disable_pressed_, deadman_pressed_, brake_pressed_, boost_active_, zero_twist_published_;
   int wait_for_connection_; /**< Time to wait for enable/disable topics in seconds (-1 to not wait). **/
   ros::Timer timer_;
+  
+  dynamic_reconfigure::Server<gopher_core_msgs::BagJoystickConfig> *             dynamic_reconfigure_server;
+  dynamic_reconfigure::Server<gopher_core_msgs::BagJoystickConfig>::CallbackType dynamic_reconfigure_callback;
```
Comment (stonier) [inline yocs_joyop/src/joyop.cpp]: Does this actually need to be a class variable? It never goes beyond the scope of the constructor.

**Comment by stonier** [comment]
Oh damn...I'm sorry. Open `yocs` should not depend on closed `gopher`. Aside from being a strange dependency, we can't then release it.

You'll have to move the dynamic reconfiguration file to [yocs_msgs](https://github.com/yujinrobot/yocs_msgs)....that should have actually been obvious ;)

File: yocs_joyop/src/joyop.cpp [hunk 3]
```diff
@@ -11,6 +11,8 @@
 #include <std_msgs/String.h>
 #include "boost/thread/mutex.hpp"
 #include "boost/thread/thread.hpp"
+#include <dynamic_reconfigure/server.h>
+#include <gopher_core_msgs/BagJoystickConfig.h>
```
Comment (stonier) [inline yocs_joyop/src/joyop.cpp]: This should be called `JoystickConfig`, not `BagJoystickConfig`. The configuration actually has nothing to do with bagging at all. Bagging is just an example user.

**Comment by haryngod** [comment]
@stonier I've fix them.

**Comment by haryngod** [comment]
Related PR:
-  [ yocs_msgs pull / 8](https://github.com/yujinrobot/yocs_msgs/pull/8)

**Comment by stonier** [comment]
```
     _  _(o)_(o)_  _
   ._\`:_ F S M _:' \_,
       / (`---'\ `-.
    ,-`  _)    (_,
```

**Pull request merged**